chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -742,6 +744,61 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
         Ok(())
     }
 
+    fn check_data_deposit_outputs(&self, block: &Block) -> Result<(), CheckBlockTransactionsError> {
+        let prev_block_id = block.prev_block_id();
+        let current_height = self
+            .get_gen_block_index(&prev_block_id)?
```

Comment:
there is `get_previous_block_index`

Reply:
I don't have the current block index here. Besides, this code will just go later.